src/lib/mina/actions/offchain-contract.unit-test.ts

```diff
@@ -140,16 +159,161 @@ await testLocal(
     // settle
     async () => {
       Local.resetProofsEnabled();
-      console.time('settlement proof 2');
-      let proof = await offchainState.createSettlementProof();
-      console.timeEnd('settlement proof 2');
+      let proof = await contract.offchainState.createSettlementProof();
 
       return transaction('settle 2', () => contract.settle(proof));
     },
 
     // check balance and supply
-    expectState(offchainState.fields.totalSupply, 1555n),
-    expectState(offchainState.fields.accounts, [sender, 900n]),
-    expectState(offchainState.fields.accounts, [receiver, 100n]),
+    expectState(contract.offchainState.fields.totalSupply, 1555n),
+    expectState(contract.offchainState.fields.accounts, [sender, 900n]),
+    expectState(contract.offchainState.fields.accounts, [receiver, 100n]),
   ]
 );
+
+// Test with multiple instances of the conract and offchain state deployed on the same network
+
+const {
+  offchainStateInstance: offchainStateInstanceA,
+  ExampleContract: ExampleContractA,
+} = ExampleContractWithState();
+
+const {
+  offchainStateInstance: offchainStateInstanceB,
+  ExampleContract: ExampleContractB,
+} = ExampleContractWithState();
+
+const Local = await Mina.LocalBlockchain({ proofsEnabled: true });
```

Comment:
`testLocal` accepts a single contract instance only, so I re-implemented the behavior here for the specific case I needed.